app/utils/repo_fetcher.py

```
@@ -0,0 +1,54 @@
+from typing import List, Protocol, Tuple
+
+from app.config import GitHosting
+from app.schemas.repo import GitRepoResponse, GitRepoResponseTransformer
+from app.utils.github_manager import GitHubManager
+from app.utils.gitlab_manager import GitLabManager
+
+
+class IRepoFetcher(Protocol):
```

Pylint (3.3.7)

[refactor] Too few public methods (1/2)

(R0903)